File: src/main/scala/euler/problems/Problem_0001.scala
package euler.problems

/*
 * If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
 *
 * Find the sum of all the multiples of 3 or 5 below 1000.
 */
object Problem_0001 extends Problem {
  
  def solveFor(limit: Int) = { 
    for {
      x <- 1 until limit
      if x % 3 == 0 || x % 5 == 0
    } yield x
  } sum
  
  def answer = solveFor(1000)

}

File: src/main/java/org/projecteuler/Problem_0001.java
package org.projecteuler;

/**
 * If we list all the natural numbers below 10 that are multiples of 3 or 5,
 * we get 3, 5, 6 and 9. The sum of these multiples is 23.
 *
 * Find the sum of all the multiples of 3 or 5 below 1000.
 *
 * @author Stefan Lourens <stefan.lourens@gmail.com>
 */
public class Problem_0001 {

    private static int solve() {
        int sum = 0;

        for (int i = 0; i < 1000; i++) {
            if (i % 3 == 0) {
                sum += i;
            } else if (i % 5 == 0) {
                sum += i;
            }
        }

        return sum;
    }

    public static void main(String ... args) {
        long start = System.currentTimeMillis();
        System.out.println(solve());
        System.out.println(System.currentTimeMillis() - start);
    }
}
